### PR TITLE
New warnings fix Solidity

### LINKED ISSUE #692: Fix compiler warnings
Right now we have some compiler [warnings](https://github.com/ubiquity/ubiquity-dollar/actions/runs/5321726749/jobs/9637204382)

What should be done:
1. Fix compiler warnings
2. Make sure [build and test workflow](https://github.com/ubiquity/ubiquity-dollar/blob/development/.github/workflows/build-and-test.yml) fails on any new warning introduced

### PR DESCRIPTION
**Pull request by molecula451**
Resolves #692

Originally done at #704 

Original requirements by @rndquu 

> Right now we have some compiler [warnings](https://github.com/ubiquity/ubiquity-dollar/actions/runs/5321726749/jobs/9637204382)
> 
> What should be done:
> 
> 1. Fix compiler warnings

### This PR fixes all compiler warnings!

https://github.com/ubiquity/ubiquity-dollar/assets/41552663/d78fde4d-0caa-477f-93d8-a4b5cdb0cec9

> 2. Make sure [build and test workflow](https://github.com/ubiquity/ubiquity-dollar/blob/development/.github/workflows/build-and-test.yml) fails on any new warning introduced

### This PR Indeed! makes the workflow 'Build and Test' fail when any new warning introduced

there is a CI build here that shows how the builds fails https://github.com/molecula451/ubiquity-dollar/actions/runs/5355653683/jobs/9714193621

**But there is more.** This PR comes with an update to foundry on the CI flow necessary to match the EVM version for the new update impossible without updating the builds

### CODE REVIEW

**Comment by molecula451** [comment]
Please merge

**Comment by molecula451** [comment]
@rndquu merge

**Comment by rndquu** [comment]
@pavlovcik pls check this PR again

This is basically this [PR](https://github.com/ubiquity/ubiquity-dollar/pull/704) (approved by all reviewers) but with the latest changes from the `development` branch

We better merge it because every new update in the `development` branch requires refactor in this PR

The "Compare Test Coverage" workflow fails but there are only slight refactors so we can skip this check

**Review by rndquu — APPROVED** [review]

**Comment by molecula451** [comment]
had to clean the PR due to other branches conflcits when new commits happened today, good to push @pavlovcik

File: packages/contracts/src/dollar/DirectGovernanceFarmer.sol
Comment (0x4007) [inline packages/contracts/src/dollar/DirectGovernanceFarmer.sol:80]: Seems unusual to me that these are all commented out.

**Review by 0x4007 — COMMENTED** [review]
Why not delete the commented out identifiers?

**Comment by molecula451** [comment]
> Why not delete the commented out identifiers?

these are not commented out, in Solidity, the variables that uses /* var */ means that it preserves the pattern of the inherit contract, to use whenever is able to do it. You cannot just remove it, at this is part of a pattern that its being use in child contracts

File: packages/contracts/src/dollar/DirectGovernanceFarmer.sol
Comment (molecula451) [reply packages/contracts/src/dollar/DirectGovernanceFarmer.sol:80]: these are not commented out. This pattern says to the contract to use the var only when available on a child contract

Comment (molecula451) [reply packages/contracts/src/dollar/DirectGovernanceFarmer.sol:80]: also these are the same changes than the last one! with only new push us to slightly changes due to new commits on development. but you may ask me anything @pavlovcik. but please go on, and then merge because the last commit broke the whole PR

Comment (molecula451) [reply packages/contracts/src/dollar/DirectGovernanceFarmer.sol:80]: this is a comment in solidity 

`// Comment`

`/* var */` this is a pattern that preservers contracts pattern to be use or that are being use in child contract

**Review by 0x4007 — APPROVED** [review]
Seems to check out. Waiting on @zgorizzo69's approval then.

**Comment by molecula451** [comment]
> Seems to check out. Waiting on @zgorizzo69's approval then.

this is the same #704 that you guys already checked out, should be merge, new commits often break things, had to close #704 for that reason

**Comment by molecula451** [comment]
@zgorizzo69

**Comment by 0x4007** [comment]
> > Seems to check out. Waiting on @zgorizzo69's approval then.
> 
> this is the same #704 that you guys already checked out, should be merge, new commits often break things, had to close #704 for that reason

For various reasons I think it's best to fully review every pull request before its merged, regardless of how similar it is to another pull request.

**Comment by molecula451** [comment]
> We better merge it because every new update in the `development` branch requires refactor in this PR

I agree but please consider rndquu words

**Comment by molecula451** [comment]
> > > Seems to check out. Waiting on @zgorizzo69's approval then.
> > 
> > 
> > this is the same #704 that you guys already checked out, should be merge, new commits often break things, had to close #704 for that reason
> 
> For various reasons I think it's best to fully review every pull request before its merged, regardless of how similar it is to another pull request.

as this is a PR that tackles compiler warnings, people introducing new code to contracts may trigger more warnings, from now after the PR people introducing code to contracts, should care about compiling warnings.

Reason that I had to close a clean PR at #704

**Comment by 0x4007** [comment]
We can consider relying on a retroactive review from @zgorizzo69

That can be handled after this is merged in.

**Comment by molecula451** [comment]
> We can consider relying on a retroactive review from @zgorizzo69
> 
> That can be handled after this is merged in.

💯,

**Comment by rndquu** [comment]
To sum up:
- this PR is basically a copy of https://github.com/ubiquity/ubiquity-dollar/pull/704 which is already approved by 3 reviewers
- this PR introduces only minor changes 
- there are no compiler warnings anymore (otherwise the "build and test" workflow would fail)

There are 2 failing workflows:
1. `Slither Analysis` (this is introduced in the latest `development` branch, we'll fix it)
2. `coverage/coveralls` (coverage dropped by 0.02% which is negligible)

I think this PR is good to merge keeping in mind that every new update in the `development` branch requires refactor in this PR

**Comment by molecula451** [comment]
Thanks for the eye opener to the team @rndquu

**Pull request merged**
